cloud_functions/climateiq_data_export_cf/main.py

```diff
@@ -20,34 +22,59 @@ def export_model_predictions(cloud_event: CloudEvent) -> pd.DataFrame:
     Args:
         cloud_event: The CloudEvent representing the storage event. The name
         of the object should conform to the following pattern:
-        "<prediction_type>/<model_id>/<study_area_name><scenario_id>/<chunk_id>"
+        "<prediction_type>/<model_id>/<study_area_name>/<scenario_id>/<chunk_id>"
     Returns:
         A DataFrame containing the lat/lon coordinates of cell centers in a
-        single chunk, representing a subset of the full study area results.
+        single chunk along with associated predictions, representing a
+        subset of the full study area results.
     Raises:
         ValueError: If the object name format, study area metadata or chunk
         area metadata is invalid.
     """
     data = cloud_event.data
-    name = data["name"]
+    object_name = data["name"]
+    bucket_name = data["bucket"]
 
     # Extract components from the object name.
-    path = pathlib.PurePosixPath(name)
+    path = pathlib.PurePosixPath(object_name)
     if len(path.parts) != 5:
         raise ValueError("Invalid object name format. Expected 5 components.")
 
     prediction_type, model_id, study_area_name, scenario_id, chunk_id = (
         path.parts
     )
 
+    predictions = _read_chunk(bucket_name, object_name)
     study_area_metadata = _get_study_area_metadata(study_area_name)
     chunk_metadata = _get_chunk_metadata(study_area_metadata, chunk_id)
 
     return _build_spatialized_model_predictions(
-        study_area_metadata, chunk_metadata
+        study_area_metadata, chunk_metadata, predictions
     )
 
 
+# TODO: Modify this logic once CNN output schema is confirmed. Also update to
+# account for errors and special values.
+def _read_chunk(bucket_name: str, object_name: str) -> np.ndarray:
```

Comment:
from the documentation
https://cloud.google.com/vertex-ai/docs/predictions/get-batch-predictions
the resulting file is going to be one line per prediction, e.g. from the docs
```
{ "instance": [1, 2, 3, 4], "prediction": [0.1,0.9]}
{ "instance": [5, 6, 7, 8], "prediction": [0.7,0.3]}
```
so we'd need to iterate over each line in the file to get one prediction per line